mobile/docker/Dockerfile

```
@@ -1,73 +1,264 @@
+
 RUN PROTOC_SHA256="75d8a9d7a2c42566e46411750d589c51276242d8b6247a5724bac0f9283e05a8" \
- && PROTOC_TARBALL="protoc-3.20.0-linux-x86_64.zip" \
- && wget -q "https://github.com/protocolbuffers/protobuf/releases/download/v3.20.0/${PROTOC_TARBALL}" \
- && echo "${PROTOC_SHA256} ${PROTOC_TARBALL}" | sha256sum -c \
- && sudo unzip -d /usr/local "${PROTOC_TARBALL}" \
- && rm "${PROTOC_TARBALL}"
+    && PROTOC_TARBALL="protoc-${PROTOC_VERSION}-linux-x86_64.zip" \
+    && wget -q "https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOC_VERSION}/${PROTOC_TARBALL}" \
+    && echo "${PROTOC_SHA256} ${PROTOC_TARBALL}" | sha256sum -c \
+    && unzip -q -d /usr/local "${PROTOC_TARBALL}" \
+    && rm "${PROTOC_TARBALL}"
 
-# Install Protoc-deg-go
 RUN PROTOC_GEN_SHA256="0b2c257938a8cd9ba3506bbdbbaad45e51245b6f9e0743035ade7acf746c6be7" \
- && PROTOC_GEN_TARBALL="protoc-gen-go.v1.34.1.linux.amd64.tar.gz" \
- && wget -q "https://github.com/protocolbuffers/protobuf-go/releases/download/v1.34.1/${PROTOC_GEN_TARBALL}" \
- && echo "${PROTOC_GEN_SHA256} ${PROTOC_GEN_TARBALL}" | sha256sum -c \
- && sudo tar -C /usr/local/bin -xzf "${PROTOC_GEN_TARBALL}" \
- && rm "${PROTOC_GEN_TARBALL}"
+    && PROTOC_GEN_TARBALL="protoc-gen-go.v1.34.1.linux.amd64.tar.gz" \
+    && wget -q "https://github.com/protocolbuffers/protobuf-go/releases/download/v1.34.1/${PROTOC_GEN_TARBALL}" \
+    && echo "${PROTOC_GEN_SHA256} ${PROTOC_GEN_TARBALL}" | sha256sum -c \
+    && tar -C /usr/local/bin -xzf "${PROTOC_GEN_TARBALL}" \
+    && rm "${PROTOC_GEN_TARBALL}"
+
+# Set PATH for Android SDK tools
+ENV PATH="/opt/android-sdk/cmdline-tools/latest/bin:/opt/android-sdk/platform-tools:/usr/local/go/bin:${PATH}"
+
+USER jenkins
+WORKDIR /home/jenkins
+
+# Accept licenses and install Android components
+RUN yes | sdkmanager --licenses >/dev/null 2>&1 \
+    && sdkmanager "platform-tools" "build-tools;${ANDROID_BUILD_TOOLS}" "platforms;${ANDROID_PLATFORM}" \
+    && sdkmanager "ndk;${NDK_VERSION}"
 
-# Install Go tools needed for the build
 RUN go install github.com/go-bindata/go-bindata/v3/go-bindata@latest \
- && go install go.uber.org/mock/mockgen@v0.4.0
+    && go install go.uber.org/mock/mockgen@v0.4.0
+
+RUN python3 -m venv /home/jenkins/aqt-venv \
+    && /home/jenkins/aqt-venv/bin/pip install --upgrade pip \
+    && /home/jenkins/aqt-venv/bin/pip install git+https://github.com/miurahr/aqtinstall.git@v3.3.0
+
+ENV PATH="/home/jenkins/aqt-venv/bin:/home/jenkins/go/bin:${PATH}"
+
+RUN echo "=== Installing Qt Desktop ===" \
+    && aqt install-qt linux desktop ${QT_VERSION} linux_gcc_64 \
+        --outputdir /opt/qt \
+       --modules qt5compat \
+    && echo "Qt Desktop installation completed"
+
+RUN echo "=== Installing Qt Android arm64-v8a ===" \
+    && aqt install-qt linux android ${QT_VERSION} android_arm64_v8a \
+        --outputdir /opt/qt \
+        --modules qtimageformats qtmultimedia qtwebview qt5compat \
+    && echo "Qt Android installations completed"
```

This could be simplified a bit

```suggestion
RUN echo "=== Installing Qt Android arm64-v8a ===" \
    && aqt install-qt linux android ${QT_VERSION} android_arm64_v8a \
        --outputdir /opt/qt \
        --modules qtimageformats qtmultimedia qtwebview qt5compat \
        --autodesktop \
    && echo "Qt Android installations completed"
```